data/scripts/lib/register_monster_type.lua

```diff
@@ -335,6 +335,9 @@ registerMonsterType.attacks = function(mtype, mask)
 					if attack.shootEffect then
 						spell:setCombatShootEffect(attack.shootEffect)
 					end
+					if attack.name == "drunk" and attack.drunkenness then
+						spell:setConditionDrunkenness(attack.drunkenness)
+					end
```

Comment:
Now that drunk is a condition the implementation of attacks for it in both xml and RevScriptSys feels inconsistent.

Should a "drunkcondition" attack be accepted for XML monsters to be consistent with other conditions?

On the RevScriptSys side, this PR requires `attack type` to now be set to condition_drunk for drunk attacks which is breaking change for RevScriptSys, not that many people use it. I just as much rather see the check be changed to `if attack.drunkenness then` and require attack type be set correctly but I know other devs here are pretty picky about that stuff. To be backwards compatible can we do something like this? 

```suggestion
					if attack.name == "drunk" or (attack.drunkenness and attack.type == CONDITION_DRUNK) then
     					spell:setConditionType(CONDITION_DRUNK)
						spell:setConditionDrunkenness(attack.drunkenness)
					end
```

The current way to cause drunk would continue working, and to cause drunk using `condition` attacks would work.

```
monster.attacks = {
	{name = "drunk", interval = 2000, chance = 10, range = 7, duration = 5000}
}
```
and
```
monster.attacks = {
	{name = "condition", type = CONDITION_DRUNK, interval = 2000, chance = 10, range = 7, duration = 5000, drunkenness = 80}
}
```

Reply:
I doubled checked this comment but failed to understand it at first.

I have now checked again the file `data/monster/lua/#example.lua` and you are right:
```lua
monster.attacks = {
	{name = "drunk", interval = 2000, chance = 10, range = 7, duration = 5000}
}
```
This should be the right way to declare a `drunk` type of attack. I thought it should follow the same syntax as in the XML files. I'll update the branch right away.

However, in my opinion the LUA logic must be:
```lua
if attack.name == "drunk" then
	spell:setConditionType(CONDITION_DRUNK)
	-- it is not mandatory to set a drunkenness value, by default it falls back to a 25% value
	if attack.drunkenness then
		spell:setConditionDrunkenness(attack.drunkenness)
	end
end
```

Reply:
@amatria if you wish. You can update it at a later date, that would be very helpful.